tracer/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -1178,6 +1211,139 @@ bool CorProfiler::ProfilerAssemblyIsLoadedIntoAppDomain(AppDomainID app_domain_i
+    //
+    mdAssemblyRef managed_profiler_assemblyRef;
+    hr = module_metadata.assembly_emit->DefineAssemblyRef(
+        managed_profiler_assembly_property.ppbPublicKey, managed_profiler_assembly_property.pcbPublicKey,
+        managed_profiler_assembly_property.szName.data(), &managed_profiler_assembly_property.pMetaData,
+        &managed_profiler_assembly_property.pulHashAlgId, sizeof(managed_profiler_assembly_property.pulHashAlgId),
+        managed_profiler_assembly_property.assemblyFlags, &managed_profiler_assemblyRef);
+
+    if (FAILED(hr) || managed_profiler_assemblyRef == mdAssemblyRefNil)
+    {
+        Logger::Warn("Error rewriting for Distributed Tracing on getting ManagedProfiler AssemblyRef");
+        return hr;
+    }
+
+    mdTypeRef distributedTracerTypeRef;
+    hr = module_metadata.metadata_emit->DefineTypeRefByName(
+        managed_profiler_assemblyRef, WStr("Datadog.Trace.ClrProfiler.DistributedTracer"), &distributedTracerTypeRef);
+
+    if (FAILED(hr))
+    {
+        Logger::Warn("Error rewriting for Distributed Tracing on getting DistributedTracer TypeRef");
+        return hr;
+    }
+
+    //
+    // *** GetDistributedTrace MethodDef ***
+    //
+    COR_SIGNATURE getDistributedTracerSignature[] = {IMAGE_CEE_CS_CALLCONV_DEFAULT, 0, ELEMENT_TYPE_OBJECT};
+    mdMethodDef getDistributedTraceMethodDef;
+    hr = module_metadata.metadata_import->FindMethod(distributedTracerTypeDef, WStr("GetDistributedTracer"),
+                                                     getDistributedTracerSignature, 3, &getDistributedTraceMethodDef);
+    if (FAILED(hr))
+    {
+        Logger::Warn("Error rewriting for Distributed Tracing on getting GetDistributedTracer MethodDef");
+        return hr;
+    }
+
+    //
+    // *** GetDistributedTrace MemberRef ***
+    //
+    mdMemberRef getDistributedTraceMemberRef;
+    hr =
+        module_metadata.metadata_emit->DefineMemberRef(distributedTracerTypeRef, WStr("GetDistributedTracer"),
+                                                       getDistributedTracerSignature, 3, &getDistributedTraceMemberRef);
+    if (FAILED(hr))
+    {
+        Logger::Warn("Error rewriting for Distributed Tracing on defining GetDistributedTracer MemberRef");
+        return hr;
+    }
+
+    ILRewriter getterRewriter(this->info_, nullptr, module_id, getDistributedTraceMethodDef);
+    getterRewriter.InitializeTiny();
+
+    // Modify first instruction from ldnull to call
+    ILRewriterWrapper getterWrapper(&getterRewriter);
+    getterWrapper.SetILPosition(getterRewriter.GetILList()->m_pNext);
+    getterWrapper.CallMember(getDistributedTraceMemberRef, false);
+    getterWrapper.Return();
+
+    hr = getterRewriter.Export();
+
+    if (FAILED(hr))
+    {
+        Logger::Warn("Error rewriting GetDistributedTracer->[AutoInstrumentation]GetDistributedTracer");
+        return hr;
+    }
+
+    Logger::Info("Rewriting GetDistributedTracer->[AutoInstrumentation]GetDistributedTracer");
```

Comment:
nit: the logging at the end of this function could be moved up so no matter what happens with the exporting of the IL, we can easily see what our rewriting did